concordium-consensus/src/Concordium/GlobalState/BlockState.hs

```diff
@@ -440,7 +569,39 @@ class (BlockStateQuery m) => BlockStateOperations m where
   --
   -- The caller MUST ensure that the staked amount does not exceed the total
   -- balance on the account.
-  bsoAddBaker :: UpdatableBlockState m -> AccountIndex -> BakerAdd -> m (BakerAddResult, UpdatableBlockState m)
+  bsoAddBaker
+    :: (AccountVersionFor (MPV m) ~ 'AccountV0, ChainParametersVersionFor (MPV m) ~ 'ChainParametersV0)
+    => UpdatableBlockState m
+    -> AccountIndex
+    -> BakerAdd
+    -> m (BakerAddResult, UpdatableBlockState m)
+
+  -- |From chain parameters version >= 1, this operation is used to add/remove/update a baker.
+  -- When adding baker, it is assumed that 'AccountIndex' account is NOT a baker and NOT a delegator.
+  bsoConfigureBaker
+    :: (AccountVersionFor (MPV m) ~ 'AccountV1, ChainParametersVersionFor (MPV m) ~ 'ChainParametersV1)
+    => UpdatableBlockState m
+    -> AccountIndex
+    -> BakerConfigure
+    -> m (BakerConfigureResult, UpdatableBlockState m)
+  
+  -- |Constrain the baker's commission rates to fall in the given ranges.
+  -- If the account is invalid or not a baker, this does nothing.
+  bsoConstrainBakerCommission
+    :: (AccountVersionFor (MPV m) ~ 'AccountV1, ChainParametersVersionFor (MPV m) ~ 'ChainParametersV1)
+    => UpdatableBlockState m
+    -> AccountIndex
+    -> CommissionRanges
+    -> m (UpdatableBlockState m)
+
+  -- |From chain parameters version >= 1, this operation is used to add/remove/update a delegator.
+  -- When adding delegator, it is assumed that 'AccountIndex' account is NOT a baker and NOT a delegator.
+  bsoConfigureDelegation
```

Comment:
I think that marking the `UpdateBlockState m` argument the last one in all block operations would help readability.

If it were the case, a block operation partially applied to all other arguments would have the signature `UpdatableBlockState m -> m (UpdatableBlockState m)`. Then, instead of naming intermediate block states (with uninformative names like `bs0`, `bs1`, `bs2, etc) in composite block operations (such as `doMinting`, `doBlockReward`), one could simply compose the primitive operations with monadic binds.

Yes, this change would be a purely stylistic one, but in my opinion, it's better to write a composition as a composition outright rather than let the reader obtain that insight.

See the comment on `doBlockRewardP4` for an example.